tests/test_mutualauth.py

```diff
@@ -0,0 +1,55 @@
+import pytest
+import unittest
+
+import falcon
+from falcon_mutualauth.mutualauth import Authorize
+
+
+class FakeRequest(object):
+
+    def __init__(self, role):
+        self.__role = role
+        self.env = {'REMOTE_ADDR': 'http://test.test'}
+        self.user_agent = 'pytest'
+
+    def get_header(self, header):
+        assert header == 'X-User-Roles'
+        return self.__role
+
+
+class FakeResource(object):
+
+    def __init__(self, authorized_for):
+        self.authorized_for = authorized_for
+
+
+class TestAuthorize(unittest.TestCase):
+
+    def setUp(self):
+        self._auth = Authorize('admin', 'principle')
+        self._resource = self._create_fake_resource()
+
+    def _create_fake_resource(self):
+        return None
+
+    def test_log_in_as_admin(self):
+        self._auth.process_resource(FakeRequest('admin'), None, self._resource)
```

Comment:
I'd be ok with that - then the test that has code using a real object returned by `_create_fake_resource` would be the only one declaring it. Further, it would allow `TestAuthorizeWithResource` to not need to inherit from this test. It could have its own `setUp` method where `self._resource` is set.